dnf5/context.cpp

```diff
@@ -560,36 +560,29 @@ std::vector<std::string> match_specs(
         installed = available = false;
     }
 
-    if (installed) {
-        try {
-            base.get_repo_sack()->get_system_repo()->load();
-            base.get_rpm_package_sack()->load_config_excludes_includes();
-        } catch (...) {
-            // Ignores errors when completing installed packages, other completions may still work.
-        }
-    }
-
-    if (available) {
-        try {
+    try {
+        libdnf5::repo::RepoQuery enabled_repos(base);
+        enabled_repos.filter_enabled(true);
+        enabled_repos.filter_type(libdnf5::repo::Repo::Type::AVAILABLE);
+        if (available) {
             // create rpm repositories according configuration files
             base.get_repo_sack()->create_repos_from_system_configuration();
             base.get_config().get_optional_metadata_types_option().set(
                 libdnf5::Option::Priority::RUNTIME, libdnf5::OptionStringSet::ValueType{});
 
             ctx.apply_repository_setopts();
 
-            libdnf5::repo::RepoQuery enabled_repos(base);
-            enabled_repos.filter_enabled(true);
-            enabled_repos.filter_type(libdnf5::repo::Repo::Type::AVAILABLE);
             for (auto & repo : enabled_repos.get_data()) {
                 repo->set_sync_strategy(libdnf5::repo::Repo::SyncStrategy::ONLY_CACHE);
                 repo->get_config().get_skip_if_unavailable_option().set(libdnf5::Option::Priority::RUNTIME, true);
             }
-
-            ctx.load_repos(false);
-        } catch (...) {
-            // Ignores errors when completing available packages, other completions may still work.
+        } else {
+            std::for_each(enabled_repos.begin(), enabled_repos.end(), [](auto & repo) { repo->disable(); });
```

Comment:
As we are already introducing breaking changes, what about changing the signature of RepoSack::update_and_load_enabled_repos to accept a single parameter with an enum specifying which repositories we want to load: only system / only available / all (default)? This way, users or other methods don't have to execute the repos disable method in case only the system repo is wanted.